### ISSUE TITLE
How to access data in collections?

### ISSUE THREAD

**Issue by flosse** [cobalt-org/cobalt.rs#557]
I want to access the `post.data.image` (in line `14`) variable but that doesn't seem to work :(

```liquid
 10             {% for post in collections.posts.pages %}
 11               <section class="12u 24u(narrower)">
 12                 <div class="box post">
 13                   <a class="image left" href={{ post.permalink }}>
 14                     <img src={{post.data.image}}></img>                                                                                                                         
 15                   </a>
 16                   <div class="inner">
 17                     <a href={{ post.permalink }}><h3>{{ post.title }}</h3></a>
 18                     <p class="date">{{ post.published_date | date: "%d. %m. %Y" }}</p>
 19                     <p>{{ post.excerpt }}<a href={{ post.permalink }}>weiter lesen ...</a>
 20                     </p>
 21                   </div>
 22                 </div>
 23               </section>
 24             {% endfor %}
```

**Comment by epage** [comment]
I assume you get `Invalid index`?

I assume you are trying to access the following in your frontmatter?
```yaml
---
...
data:
  image: ...
---
...

**Comment by epage** [comment]
btw in an unreleased version of liquid, the error will now list what indexes are available which should help.  I'm wanting to wait on a PR before getting it bubbled up into a cobalt release.

**Comment by flosse** [comment]
The frontmatter looks like this:
```
  1 ---
  2 title: "Rust in der Industrie & Automatisierung"
  3 published_date: "2018-10-09 00:00:00 +0100"
  4 layout: post.liquid
  5 data:
  6   tags: "stammtisch,rust,industrie,iot,i40"
  7   image: /images/rust.png
  8   lang: de
  9   route: /blog
 10 ---
```

**Comment by flosse** [comment]
That's the error:
```
Error: build command failed
Caused by: Failed to render for "blog.html"
Caused by: Failed to render layout "posts.liquid"
Caused by: liquid: Unknown index
  with:
    variable=post.data.image
    index=image
from: {% for post in collections["posts"]["pages"]  %}
  with:
    post=published_date: 2018-01-02 00:00:00 +0000, previous: categories: , is_draft: false, collection: posts, file: parent: posts, permalink: posts/2017-11-26-licht-fuer-kinder-in-afrika.md, slug: licht-fuer-kinder-in-afrika, data: image: /images/schule_in_afrika.jpg, excerpt: Ein Projekt zur Verbreitung erneuerbarer Energien in Afrika. , lang: de, tags: licht, afrika, solar, bildung, route: /blog, published_date: 2017-11-26 00:00:00 +0100, title: Licht und Perspektiven für Kinder in Afrika, permalink: posts/2017-11-26-licht-fuer-kinder-in-afrika.html, description: , is_draft: false, collection: posts, categories: , title: Alles Gute für 2018 !, data: data: lang: de, tags: stammtisch, diskussion, kennenlernen, route: /blog, image: /images/erster_stammtisch_2018.jpg, layout: post.liquid, published_date: 2018-01-02 00:00:00 +0100, permalink: posts/2018-01-02-alles-gute-fuer-2018.html, slug: alles-gute-fuer-2018, file: parent: posts, permalink: posts/2018-01-02-alles-gute-fuer-2018.md,
```

**Comment by epage** [comment]
I tried to reproduce this but I can't get the error with https://github.com/epage/cobalt-issue-557

Could you clone that and see what changes are needed before you can get it t reproduce?

**Comment by epage** [comment]
I noticed that in the error, the `post` does not include `excerpt`.

Since I haven't touched this part in a while, I had to dig in further.  Apparently, `content` and `excerpt` are only available for your own page during the processing of posts.  Once we start processing non-post pages, we make these variables available.

I'm assuming the intent is for creating a menu.  We'd need to split layout rendering into a separate pass from content/excerpt.  This seems like a reasonable request until / if we get support for proper menu generation.
